docs/index.md

```diff
@@ -7,20 +13,28 @@ The provider needs to be configured with the proper credentials before it can be
 
 * RouterOS v6.45.2+ (It may work with other versions but it is untested against other versions!)
 
-## Using the provider
-
-This provider is on the terraform registry so you only need to reference it in your terraform code (example below).
 
 ## Example Usage
-
-```hcl
+```terraform
 # Configure the mikrotik Provider
 provider "mikrotik" {
-  host = "hostname-of-server:8728"     # Or set MIKROTIK_HOST environment variable
-  username = "<username>"              # Or set MIKROTIK_USER environment variable
-  password = "<password>"              # Or set MIKROTIK_PASSWORD environment variable
-  tls = true|false                     # Or set MIKROTIK_TLS environment variable
-  ca_certificate = "/path/to/ca/certificate.pem"    # Or set MIKROTIK_CA_CERTIFICATE environment variable
-  insecure = true|false                # Or set MIKROTIK_INSECURE environment variable
+  host           = "hostname-of-server:8728"     # Or set MIKROTIK_HOST environment variable
+  username       = "<username>"                  # Or set MIKROTIK_USER environment variable
+  password       = "<password>"                  # Or set MIKROTIK_PASSWORD environment variable
+  tls            = true                          # Or set MIKROTIK_TLS environment variable
+  ca_certificate = "/path/to/ca/certificate.pem" # Or set MIKROTIK_CA_CERTIFICATE environment variable
+  insecure       = true                          # Or set MIKROTIK_INSECURE environment variable
 }
 ```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `ca_certificate` (String) Path to MikroTik's certificate authority
+- `host` (String) Hostname of the MikroTik router
+- `insecure` (Boolean) Insecure connection does not verify MikroTik's TLS certificate
+- `password` (String) Password for MikroTik api
+- `tls` (Boolean) Whether to use TLS when connecting to MikroTik or not
+- `username` (String) User account for MikroTik api
```

Comment:
username, password and host are not optional fields. So it seems like the provider's arguments aren't set correctly?

Reply:
that's how it is [declared](https://github.com/ddelnano/terraform-provider-mikrotik/blob/543dbda3b40aec544fcad8adad77a344b9fa7cfe/mikrotik/provider.go#L22):
```go
"username": {
    Type:        schema.TypeString,
    Required:    true,
    DefaultFunc: schema.EnvDefaultFunc("MIKROTIK_USER", nil),
    Description: "User account for mikrotik api",
},
```
so yes, it should either be set to `Optional` or update the docs and mention that there is a fallback to env vars.